controllers/vaultdynamicsecret_controller.go

```diff
@@ -123,6 +123,13 @@ func (r *VaultDynamicSecretReconciler) Reconcile(ctx context.Context, req ctrl.R
 		return ctrl.Result{}, r.handleDeletion(ctx, o)
 	}
 
+	if addedFinalizer, err := maybeAddFinalizer(ctx, r.Client, o, vaultDynamicSecretFinalizer); err != nil {
+		return ctrl.Result{}, err
+	} else if addedFinalizer {
+		// the finalizer was added, requeue the request.
+		return ctrl.Result{Requeue: true}, nil
+	}
```

Comment:
There's a comment in maybeAddFinalizer():
https://github.com/hashicorp/vault-secrets-operator/blob/e024c7566022fcddfdd695127666f73663fe801f/controllers/common.go#L248-L249

Any concern with API validation errors now that maybeAddFinalizer() is being called before updateStatus()?

Reply:
Thank you for reviewing! My understanding for the comment is that an API validation error would happen if both metadata (finalizers) and status of a resource are updated in the same reconcile pass. But in our case, we are calling `maybeAddFinalizer()` first, and if a finalizer is added, we exit early and requeue. So we shouldn't be touching the status in the same loop, which would avoid the validation issue because the next reconcile handles the status update separately. Please let me know if I am misunderstanding!